x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -458,17 +456,17 @@ private LogicalPlan resolveStats(Stats stats, List<Attribute> childrenOutput) {
                 }
             }
 
-            if (stats.expressionsResolved() == false) {
-                AttributeMap<Expression> resolved = new AttributeMap<>();
```

Comment:
The role of AttributeMap is to guarantee equality - while it can deal with ordering (it's a linkedmap), it's better to have this recorded elsewhere such as a list.

Reply:
Ah, fair point; however, there shouldn't be groupings with same name id but different names, so the `mergeOutputAttributes` below should take care of deduplicating fine.

I think using the `AttributeMap` here is less obvious (I completely forgot the thing preserves iteration order), esp. since an `AttributeSet` seems to suffice.